master/internal/api_notebook.go

```diff
@@ -56,21 +56,23 @@ func (a *apiServer) GetNotebooks(
 		return nil, err
 	}
 
-	a.filter(&resp.Notebooks, func(i int) bool {
-		if err != nil {
-			return false
-		}
-		ok, serverError := command.AuthZProvider.Get().CanGetNSC(
-			ctx, *curUser, model.UserID(resp.Notebooks[i].UserId),
-			model.AccessScopeID(resp.Notebooks[i].WorkspaceId),
-		)
-		if serverError != nil {
-			err = serverError
-		}
-		return ok
-	})
+	/*
+		Expected behavior:
+		- If the user doesn't specify a workspace ID we return
+		all the permitted notebooks for the user.
+		- If they have no NSC access to any workspace we return an empty list.
+		- If the user requests a workspace that they don't have access to we respond with a 404.
+	*/
+	limitedScopes, err := command.AuthZProvider.Get().AccessibleScopes(
```

Comment:
just a heads up the args changed in https://github.com/determined-ai/determined/pull/5614, if you wanted to use that version. We should probably have a chat about that func

Reply:
I think the difference in the arguments comes from https://github.com/determined-ai/determined/pull/5614#discussion_r1062846852 I used to have it set up the way you do until I worked on this commit https://github.com/determined-ai/determined/pull/5609/commits/7b659491a2f804bc267f9f32e6b3412846571b82

Reply:
decision is that I'll change my implementation to only receive one workspace ID as opposed to a slice of IDs.